### PR TITLE
Exclude BCL libraries from Roslyn vsix

### PR DESCRIPTION
**Pull request by genlu**
This change would make sure everything shipped as part of runtime would not be deployed by us again, so we don't need to manually maintain an exclusion list.

@tmat

### CODE REVIEW

**Comment by jasonmalinowski** [comment]
@genlu Is there a binary we are shipping right now that this is excluding? Or is this just cleanup to generally make this more maintainable? Both are good don't get me wrong, but is the best test of this to compare the before/after and know what changed?

File: src/Workspaces/Remote/ServiceHub.CoreComponents/Microsoft.CodeAnalysis.Remote.ServiceHub.CoreComponents.csproj [hunk 1]
```diff
@@ -18,31 +18,32 @@
...
     <PackageReference Include="SQLitePCLRaw.bundle_green" Version="$(SQLitePCLRawbundle_greenVersion)" />
+    <!-- We use the content of these two runtime pacakges to exclude BCL library from the dependencies we need to deploy as part of Roslyn vsix. -->
+    <PackageReference Include="Microsoft.NETCore.App.Runtime.win-x64" Version="$(MicrosoftNetCoreAppPackagesVersion)" ExcludeAssets="all" PrivateAssets="all" />
+    <PackageReference Include="Microsoft.WindowsDesktop.App.Runtime.win-x64" Version="$(MicrosoftNetCoreAppPackagesVersion)" ExcludeAssets="all" PrivateAssets="all" />
   </ItemGroup>
 
-  <Target Name="PublishProjectOutputGroup" DependsOnTargets="Publish" Returns="@(_PublishedFiles)">
+  <Target Name="PublishProjectOutputGroup" DependsOnTargets="Publish" Returns="@(_PublishedFiles)">  
+    <PropertyGroup>
+      <_RuntimeLibraryDir>$(NuGetPackageRoot)Microsoft.NETCore.App.Runtime.win-x64\$(MicrosoftNetCoreAppPackagesVersion)\runtimes\win-x64</_RuntimeLibraryDir>
```
Comment (tmat) [inline src/Workspaces/Remote/ServiceHub.CoreComponents/Microsoft.CodeAnalysis.Remote.ServiceHub.CoreComponents.csproj:28]: I think there is an msbuild property that contains path to package that is automatically defined by nuget.

File: src/Workspaces/Remote/ServiceHub.CoreComponents/Microsoft.CodeAnalysis.Remote.ServiceHub.CoreComponents.csproj [hunk 2]
```diff
@@ -18,31 +18,32 @@
...
-      <_ExcludedFiles Include="$(PublishDir)\System.Runtime.CompilerServices.Unsafe.dll" />
-      <_ExcludedFiles Include="$(PublishDir)\System.Security.AccessControl.dll" />
-      <_ExcludedFiles Include="$(PublishDir)\System.Security.Cryptography.ProtectedData.dll" />
-      <_ExcludedFiles Include="$(PublishDir)\System.Security.Permissions.dll" />
-      <_ExcludedFiles Include="$(PublishDir)\System.Security.Principal.Windows.dll" />
-      <_ExcludedFiles Include="$(PublishDir)\System.Threading.Tasks.Dataflow.dll" />
-      <_ExcludedFiles Include="$(PublishDir)\System.Windows.Extensions.dll" />
+      <!-- 
+        For BCL, we want to use the version provided by the runtime in VS, not the ones from the NuGet packages. 
+        (However, we can't safely exclude shared dependencies from ServiceHub host folder: we might be referencing
```
Comment (tmat) [inline src/Workspaces/Remote/ServiceHub.CoreComponents/Microsoft.CodeAnalysis.Remote.ServiceHub.CoreComponents.csproj:43]: Extra `(`

File: src/Workspaces/Remote/ServiceHub.CoreComponents/Microsoft.CodeAnalysis.Remote.ServiceHub.CoreComponents.csproj (hunk 1 above)
Comment (genlu) [reply src/Workspaces/Remote/ServiceHub.CoreComponents/Microsoft.CodeAnalysis.Remote.ServiceHub.CoreComponents.csproj:28]: Yea, that's probably the biggest trouble I have working with msbuild - I have no clue what are available vs. what needs custom logic. Let me see if I can find this property you mentioned...

**Comment by genlu** [comment]
@jasonmalinowski this is just to clean things up to make it more maintainable. While I didn't check, I don't expect this would make any difference in the vsix, since the previous list was constructed with the same logic (by manually running a console app to compare)

**Comment by genlu** [comment]
```
PackageReference Microsoft.NETCore.App.Runtime.win-x64 has incorrect version $(MicrosoftNetCoreAppPackagesVersion)
Allowed values are $(MicrosoftNETCoreAppRuntimewinx64Version) or$(MicrosoftNETCoreAppRuntimewinx64FixedVersion) or$(RefOnlyMicrosoftNETCoreAppRuntimewinx64Version)
PackageReference Microsoft.WindowsDesktop.App.Runtime.win-x64 has incorrect version $(MicrosoftNetCoreAppPackagesVersion)
Allowed values are $(MicrosoftWindowsDesktopAppRuntimewinx64Version) or$(MicrosoftWindowsDesktopAppRuntimewinx64FixedVersion) or$(RefOnlyMicrosoftWindowsDesktopAppRuntimewinx64Version)
```

buildboss reported this error, not sure how to resolve. Since we are not actually shiping anything references these two packages, they are only used for the purpose of finding what's shipped in VS. (and I'm also planning to use them as the references to pass to crossgen2 for our .Net 6 OOP host). @tmat @jaredpar any suggestions?

**Comment by genlu** [comment]
@jasonmalinowski @tmat PTAL, thanks!

**Review by tmat — APPROVED** [review]
:shipit:

**Pull request merged**